src/transformation/visitors/modules/export.ts

```diff
@@ -46,24 +47,62 @@ function transformExportAllFrom(context: TransformationContext, node: ts.ExportD
     const moduleRequire = createModuleRequire(context, node.moduleSpecifier);
     const tempModuleIdentifier = lua.createIdentifier("____export");
 
+    const result: lua.Statement[] = [];
+
     const declaration = lua.createVariableDeclarationStatement(tempModuleIdentifier, moduleRequire);
+    result.push(declaration);
+
+    if (node.exportClause && ts.isNamespaceExport(node.exportClause)) {
+        const namespaceIdentifier = lua.createIdentifier(createSafeName(node.exportClause.name.text));
+        const namespaceDeclaration = lua.createVariableDeclarationStatement(
+            namespaceIdentifier,
+            lua.createTableExpression()
+        );
+        result.push(namespaceDeclaration);
+    }
 
     const forKey = lua.createIdentifier("____exportKey");
     const forValue = lua.createIdentifier("____exportValue");
 
-    const body = lua.createBlock([
-        lua.createAssignmentStatement(lua.createTableIndexExpression(createExportsIdentifier(), forKey), forValue),
-    ]);
+    const left =
+        node.exportClause && ts.isNamespaceExport(node.exportClause)
+            ? lua.createIdentifier(createSafeName(node.exportClause.name.text))
+            : createExportsIdentifier();
+
+    const leftAssignment = lua.createAssignmentStatement(
+        lua.createTableIndexExpression(lua.cloneIdentifier(left), lua.cloneIdentifier(forKey)),
+        forValue
+    );
+
+    const ifBody = lua.createBlock([leftAssignment]);
+
+    const ifStatement = lua.createIfStatement(
```

Comment:
Could you add a comment explaining what the purpose of this if statement is.

Reply:
Added some more comments to transformExportAll, I think it should be much easier to read now